packages/core/src/components/entity-title/entityTitle.tsx

```diff
@@ -0,0 +1,148 @@
+};
+
+export const EntityTitle: React.FC<IEntityTitleProps> = ({
+    className,
+    headingSize,
+    title,
+    icon,
+    tags,
+    subtitle,
+    titleURL,
+}) => {
+    const titleComponent = React.useMemo(() => {
+        const maybeTitleWithURL =
+            titleURL != null ? (
+                <a target="_blank" href={titleURL} rel="noreferrer">
+                    {title}
+                </a>
+            ) : (
+                title
+            );
+
+        const titleClassName = classNames(Classes.ENTITY_TITLE_TITLE);
+
+        switch (headingSize) {
+            case HeadingSize.h1:
+                return <H1 className={titleClassName}>{maybeTitleWithURL}</H1>;
+            case HeadingSize.h2:
+                return <H2 className={titleClassName}>{maybeTitleWithURL}</H2>;
+            case HeadingSize.h3:
+                return <H3 className={titleClassName}>{maybeTitleWithURL}</H3>;
+            case HeadingSize.h4:
+                return <H4 className={titleClassName}>{maybeTitleWithURL}</H4>;
+            case HeadingSize.h5:
+                return <H5 className={titleClassName}>{maybeTitleWithURL}</H5>;
+            case HeadingSize.h6:
+                return <H6 className={titleClassName}>{maybeTitleWithURL}</H6>;
+            default:
+                return <Text className={titleClassName}>{maybeTitleWithURL}</Text>;
+        }
+    }, [title, titleURL, headingSize]);
+
+    const subtitleComponent = React.useMemo(() => {
+        if (!subtitle) {
+            return null;
+        }
+
+        return (
+            <Text
+                className={classNames(
+                    Classes.TEXT_MUTED,
+                    headingSize == null && Classes.TEXT_SMALL,
+                    Classes.ENTITY_TITLE_SUBTITLE,
+                )}
+            >
+                {subtitle}
+            </Text>
+        );
+    }, [headingSize, subtitle]);
+
+    return (
+        <div className={classNames(className, Classes.ENTITY_TITLE, headingSize != null && css[headingSize])}>
+            {icon != null && (
+                <div
+                    className={classNames(Classes.ENTITY_TITLE_ICON_CONTAINER, {
+                        [Classes.ENTITY_TITLE_HAS_SUBTITLE]: subtitleComponent != null,
+                    })}
+                >
+                    <Icon icon={icon} aria-hidden={true} tabIndex={-1} className={Classes.TEXT_MUTED} />
+                </div>
+            )}
+            <div className={Classes.ENTITY_TITLE_RIGHT}>
+                <div className={Classes.ENTITY_TITLE_TOP_LINE}>
+                    {titleComponent}
+                    {tags != null && <div className={Classes.ENTITY_TITLE_TAGS_CONTAINER}>{tags}</div>}
+                </div>
+                {subtitleComponent}
+            </div>
+        </div>
+    );
+};
+
+EntityTitle.defaultProps = {};
+EntityTitle.displayName = `${DISPLAYNAME_PREFIX}.EntityTitle`;
+
+type ISkeletonEntityTitleProps = Props &
```

Comment:
we don't use `I` prefixes in types or interfaces names anymore. also we typically use `interface` instead of `type` to define props interfaces:


```suggestion
interface SkeletonEntityTitleProps extends Props {
```